src/js/actions/layers.js

```diff
@@ -790,9 +791,46 @@ define(function (require, exports) {
     selectAll.post = [_verifyLayerSelection];
 
     /**
+     * Remove the given layers from the JavaScript model. This is useful from
+     * event handlers that tell us about layers which have already been removed
+     * from the model. To delete layers from a Photoshop document, use the
+     * deleteSelected action.
+     *
+     * @param {Document} document
+     * @param {Layer|Immutable.Iterable.<Layer>} layers
+     * @param {boolean=} resetHistory
+     * @return {Promise}
+     */
+    var removeLayers = function (document, layers, resetHistory) {
+        if (layers instanceof Layer) {
+            layers = Immutable.List.of(layers);
+        }
+
+        return _getSelectedLayerIndices(document)
+            .bind(this)
+            .then(function (selectedLayerIndices) {
+                var payload = {
+                    documentID: document.id,
+                    layerIDs: collection.pluck(layers, "id"),
+                    selectedIndices: selectedLayerIndices
+                };
+
+                this.dispatch(events.document.history.nonOptimistic.DELETE_LAYERS, payload);
+
+                if (resetHistory) {
+                    return this.transfer(historyActions.resetHistory, document);
```

Comment:
Per the above-noted issue with undo/redo not being available in the menu, try...

changing this line to `return this.transfer(historyActions.queryCurrentHistory, document.id);`
(and the associated `.transfers` change)

I'm not saying this is the ideal solution, but it is more correct because it asks photoshop what the current pointer should be at, and then re-inits the  history store with its known-unknowns

This throws a warning ...which could probably be forcibly quelled with a `iKnowWhatImDoing` flag